instrumentation/log/http_output_test.go

```diff
@@ -0,0 +1,121 @@
+package log
+
+import (
+	"context"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"math/rand"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+)
+
+type httpOutputHarness struct {
+	port     uint16
+	listener net.Listener
+	router   *http.ServeMux
+	server   *http.Server
+}
+
+func newHttpHarness(handler http.Handler) *httpOutputHarness {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	port := uint16(20000 + r.Intn(40000))
```

Comment:
please use port 0 to allocate a random port; this pattern is very bad and will eventually result in flakiness.

Reply:
Done.